File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-7.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        L, R = 0, len(nums) - 1

        while L <= R:
            M = (R + L) // 2
            if nums[L] == target:
                return L
            if nums[R] == target:
                return R
            if nums[M] == target:
                return M
            if nums[M] > nums[R] and (target <= nums[R] or target > nums[M]):
                L = M + 1
            else:
                R = M - 1
        
        return -1

File: Data Structures & Algorithms/two-integer-sum/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const diffMap = new Map()

        for (let i=0; i<nums.length; i++) {
            const num = nums[i]
            const difference = target - num

            if (!diffMap.has(difference)) {
                diffMap.set(num, i)
            } else {
                return [diffMap.get(difference), i]
            }
        }
    }
}
